crates/dap/src/adapters.rs

```diff
@@ -94,89 +104,79 @@ pub struct GithubRepo {
 pub async fn download_adapter_from_github(
     adapter_name: DebugAdapterName,
     github_version: AdapterVersion,
+    file_type: DownloadedFileType,
     delegate: &dyn DapDelegate,
 ) -> Result<PathBuf> {
     let adapter_path = paths::debug_adapters_dir().join(&adapter_name);
-    let version_dir = adapter_path.join(format!("{}_{}", adapter_name, github_version.tag_name));
+    let version_path = adapter_path.join(format!("{}_{}", adapter_name, github_version.tag_name));
     let fs = delegate.fs();
 
-    let http_client = delegate
-        .http_client()
-        .ok_or_else(|| anyhow!("Failed to download adapter: couldn't connect to GitHub"))?;
-
     if !adapter_path.exists() {
-        fs.create_dir(&adapter_path.as_path()).await?;
+        fs.create_dir(&adapter_path.as_path())
+            .await
+            .context("Failed creating adapter path")?;
     }
 
-    if version_dir.exists() {
-        return Ok(version_dir);
+    if version_path.exists() {
```

Comment:
I thinking about this again and maybe we could compare version_path's checksum with what we want to download to make sure it contains the same content?

 I know I was the one that originally setup the check, but a checksum would be more full proof (At least in the future)